### PR TITLE
fix(native-filters): Update filter saving

### PR DESCRIPTION
**Pull request by simcha90**
### SUMMARY
This PR fix reloading charts after saving filters config modal.

1. Clear dataMask on saving filters
2. Getting back removed cross filters after getting filters config

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/util/activeAllDashboardFilters.ts [hunk 1]
```diff
@@ -71,6 +71,18 @@ export const findAffectedCharts = ({
   );
 };
 
+export const getRelevantDataMask = (
+  dataMask: DataMaskStateWithId,
+  filterBy: string,
+  prop?: string,
+) =>
```
Comment (villebro) [inline superset-frontend/src/dashboard/util/activeAllDashboardFilters.ts:78]: Couldn't we add the return type here?

**Review by villebro — COMMENTED** [review]
Minor typing comment

File: superset-frontend/src/dashboard/util/activeAllDashboardFilters.ts (hunk 1 above)
Comment (simcha90) [reply superset-frontend/src/dashboard/util/activeAllDashboardFilters.ts]: fixed

**Review by villebro — APPROVED** [review]
LGTM, works like a charm!

**Pull request merged**